STYLE_GUIDE.md

```diff
@@ -0,0 +1,218 @@
+
+Other than that, we mostly recommend following the [AirBnB JavaScript Style Guide](https://github.com/airbnb/javascript).
+
+## Browser and device support
+
+We aim to support recent versions of major modern browsers - i.e. Chome, Firefox, Edge, Safari, and Opera, on both MacOS and Windows. Keeping progressive enhancement in mind, we aim for decent mobile support where possible, i.e. nothing should be broken on mobile, but it's okay for mobile/older browsers to have a slightly degraded experience.
+
+## Kedro UI
+
+Kedro-Viz uses [Kedro-UI](https://quantumblacklabs.github.io/kedro-ui/), our UI component library, for generic/reusable interface components like buttons, modals, dropdowns, etc. Where possible, we recommend using elements from Kedro-UI over designing and building our own.
+
+## Accessibility
+
+We aim to meet [WCAG 2.1 Level AA](https://www.w3.org/WAI/standards-guidelines/wcag/) standards where possible, while acknowledging that this is a data visualization tool, so some criteria will be difficult to pass.
+
+Kedro-Viz should be navigable with different input devices (e.g. mouse, keyboard, and touchscreen), with obvious and visible focus states on all interactive elements. Text elements should have sufficient colour contrast and font-size to ensure readability to a minimum Level AA standard. Use [WAI-ARIA](https://www.w3.org/WAI/standards-guidelines/aria/) features where applicable to improve screenreader support.
+
+We recommend checking [axe](https://www.deque.com/axe/) and [Lighthouse](https://developers.google.com/web/tools/lighthouse) to review for potential accessibility errors if unsure.
+
+## Semantic HTML
+
+Ensure that page content is marked up with [semantic HTML elements](https://html.com/semantic-markup/) to keep it as usable and accessible as possible.
+
+Avoid:
+
+```jsx
+<div className="form">
+  <div>Contact us</div>
+  <div>Use this form to send us a message.</div>
+  <div>Email address:</div>
+  <input type="text" />
+  <div onClick={handleSubmit}>Submit</div>
+</div>
+```
+
+Prefer:
+
+```jsx
+<form onSubmit={handleSubmit}>
+  <h1>Contact us</h1>
+  <p>Use this form to send us a message.</p>
+  <label htmlFor="email">Email address:</label>
+  <input id="email" type="email" />
+  <button>Submit</button>
+</form>
+```
+
+## CSS
+
+### Class names
+
+This project uses the [BEM naming convention](http://getbem.com/), to avoid conflicts, reduce rule specificity, and make classes easier to read and understand. To impove CSS maintainability we [reduce rule specificity](https://css-tricks.com/strategies-keeping-css-specificity-low/) and avoid nesting CSS selectors.
+
+Because Kedro-Viz can be imported into other applications, we avoid ever setting global styles. We use the `pipeline-` prefix for most HTML/CSS classes, to reduce the risk of clashes in the global namespace when this project is imported into other applications. It's important never to use global element selectors or generic classes like `.header`.
```

Comment:
It's probably a good idea to specifically stress here that whilst CSS files are split up per-component and imported via JS, all CSS is still naturally global in scope. 

CSS modules are pretty common now and it's easy to forget it's not the default (it would be nice though...)